Automattic/wp-calypso
# Add Stats Standalone App for Jetpack plugins

#### Proposed Changes
The Stats App is built to refresh the Stats experience in Jetpack. The counterpart of the project is in [here](https://github.com/Automattic/jetpack/pull/27247).

## Hierarchy
```
.
└── src/
    ├── components/       ← stats app only components. For now there is only a layout component.
    ├── page-middleware/  ← page.js integration with React and everything
    ├── app.js            ← entry point
    └── routes.js         ← page.js routes
```

## Building

### Production

```bash
cd apps/stats
yarn build
```
### Development

```bash
STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev
```


#### Testing Instructions
* Run `yarn start`
* Open `http://calypso.localhost:3000/stats/day/your-site`
* Ensure all stats pages work as expected

##### Optional Steps:
* Open your local Jetpack repository
* Checkout branch `add/stats-admin-package`
* Go back to Calypso repository
* Run `cd apps/stats`
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` <- replace path
* Open `https://your-jetpack-site.com/wp-admin/admin.php?page=stats&calypso_stats=1`
* Ensure **most** pages work

#### Pre-merge Checklist






Related to #
